src/RingGrids/geodesics.jl

```diff
@@ -0,0 +1,54 @@
+const DEFAULT_RADIUS = 6371e3  # Earth mean radius in meters.
+
+"""
+    abstract type AbstractSphericalDistance end
+
+Super type of formulas to calculate the spherical distance or great-circle distance.
+To define a `NewFormula`, define `struct NewFormula <: AbstractSphericalDistance end`
+and the actual calculation as a functor
+
+    function NewFormula(lonlat1::Tuple, lonlat2::Tuple; radius=DEFAULT_RADIUS, kwargs...)
+
+assuming inputs in degrees and returning the distance in meters (or radians for radius=1).
+Then use the general interface `spherical_distance(NewFormula, args...; kwargs...)`"""
+abstract type AbstractSphericalDistance end
+struct Haversine <: AbstractSphericalDistance end
```

Comment:
introduced an `AbstractSphericalDistance` so that any formula is a subtype of it which allows us to handle a "class" of types abstractly together. See how I've defined both `lon1, lat1, lon2, lat2` as well as tupled arguments `(lon1, lat1), (lon2, lat2)` for all new formulas collectively. The actual calculation is then implemented as a function of the same name, here `Haversine` which in Julia is called a functor. It's essentially an object that also has a function defined. The object here is "empty" (there are no fields in `struct Haversine ... end`)